mesa/extra.py

```diff
@@ -59,7 +59,7 @@ def weighted_n_of(n, agentset, weight_function, rng_istance):
     #minp = min(p)
     #maxp = max(p)
     p = [i/sump for i in p]
-    return  rng_istance.choice(agentset, n, replace=False, p=p)
+    return  rng_istance.choice(agentset, int(n), replace=False, p=p)
```

Comment:
Isn't this error-prone? Maybe we should check on call, not force it to be what we want it to be.

Reply:
As you can see in numpy.random.Generator.choice [documentation](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.choice.html) the parameter size accept only int. It may happen that the size parameter here is a float number, here is the reason for the casting.

![image](https://user-images.githubusercontent.com/3855193/97548992-25066480-19d0-11eb-906c-b6e074d7c512.png)